syntax = "proto3";

package codec_protos;

service Tools {
    rpc DecodeExtrinsic (ExtrinsicRequest) returns (ExtrinsicReply) {
    }
    rpc DecodeEvent (EventRequest) returns (EventReply) {
    }
    rpc DecodeLog (LogRequest) returns (LogReply) {
    }
    rpc DecodeStorage (StorageRequest) returns (StorageReply) {
    }
    rpc RegMetadata (MetadataRequest) returns (MetadataReply) {
    }

}

message ExtrinsicRequest {
    string message = 1;
    int32 metadataVersion = 2;
}

message ExtrinsicReply {
    string message = 1;
}

message EventRequest {
    string message = 1;
    int32 metadataVersion = 2;
}

message EventReply {
    string message = 1;
}

message LogRequest {
    string message = 1;
    int32 metadataVersion = 2;
}

message LogReply {
    string message = 1;
}

message StorageRequest {
    string message = 1;
    string decoderType = 2;
}

message StorageReply {
    string message = 1;
}

message MetadataRequest {
    string message = 1;
    int32 metadataVersion = 2;
}

message MetadataReply {
    bool message = 1;
}